deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/utils/KerasLayerUtils.java

```diff
@@ -225,6 +226,24 @@ public static KerasLayer getKerasLayerFromConfig(Map<String, Object> layerConfig
             layer = new KerasInput(layerConfig, enforceTrainingConfig);
         } else if (layerClassName.equals(conf.getLAYER_CLASS_NAME_MERGE())) {
             layer = new KerasMerge(layerConfig, enforceTrainingConfig);
+        } else if (layerClassName.equals(conf.getLAYER_CLASS_NAME_ADD()) ||
+                layerClassName.equals(conf.getLAYER_CLASS_NAME_ADD())) {
+            layer = new KerasMerge(layerConfig, ElementWiseVertex.Op.Add, enforceTrainingConfig);
+        } else if (layerClassName.equals(conf.getLAYER_CLASS_NAME_SUBTRACT()) ||
+                layerClassName.equals(conf.getLAYER_CLASS_NAME_FUNCTIONAL_SUBTRACT())) {
+            layer = new KerasMerge(layerConfig, ElementWiseVertex.Op.Subtract, enforceTrainingConfig);
+        } else if (layerClassName.equals(conf.getLAYER_CLASS_NAME_AVERAGE()) ||
+                layerClassName.equals(conf.getLAYER_CLASS_NAME_FUNCTIONAL_AVERAGE())) {
+            layer = new KerasMerge(layerConfig, ElementWiseVertex.Op.Average, enforceTrainingConfig);
+        } else if (layerClassName.equals(conf.getLAYER_CLASS_NAME_MULTIPLY()) ||
+                layerClassName.equals(conf.getLAYER_CLASS_NAME_FUNCTIONAL_MULTIPLY())) {
+            layer = new KerasMerge(layerConfig, ElementWiseVertex.Op.Product, enforceTrainingConfig);
+        } else if (layerClassName.equals(conf.getLAYER_CLASS_NAME_ADD()) ||
+                layerClassName.equals(conf.getLAYER_CLASS_NAME_ADD())) {
+            layer = new KerasMerge(layerConfig, ElementWiseVertex.Op.Add, enforceTrainingConfig);
+        } else if (layerClassName.equals(conf.getLAYER_CLASS_NAME_DOT()) ||
+                layerClassName.equals(conf.getLAYER_CLASS_NAME_FUNCTIONAL_DOT())) {
+            layer = new KerasMerge(layerConfig, null, enforceTrainingConfig);
```

Comment:
If mergeMode is null, wan't this use MergeVertex, and not implement dot?

Reply:
yep, that should be concatenate... :/